src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngUIPolicy.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    /// <summary>
+    ///     Configuration parameters for the UI displayed by CNG when accessing a protected key
+    /// </summary>
+    public sealed class CngUIPolicy
+    {
+        public CngUIPolicy(CngUIProtectionLevels protectionLevel)
+            : this(protectionLevel, friendlyName: null)
+        {
+        }
+
+        public CngUIPolicy(CngUIProtectionLevels protectionLevel, string friendlyName)
+            : this(protectionLevel, friendlyName, description: null)
+        {
+        }
+
+        public CngUIPolicy(CngUIProtectionLevels protectionLevel, string friendlyName, string description)
+            : this(protectionLevel, friendlyName, description, useContext: null)
+        {
+        }
+
+        public CngUIPolicy(CngUIProtectionLevels protectionLevel, string friendlyName, string description, string useContext)
+            : this(protectionLevel, friendlyName, description, useContext, creationTitle: null)
+        {
+        }
+
+        public CngUIPolicy(CngUIProtectionLevels protectionLevel, string friendlyName, string description, string useContext, string creationTitle)
+        {
+            ProtectionLevel = protectionLevel;
+            FriendlyName = friendlyName;
+            Description = description;
+            UseContext = useContext;
+            CreationTitle = creationTitle;
+        }
+
+        /// <summary>
+        ///     Level of UI protection to apply to the key
+        /// </summary>
+        public CngUIProtectionLevels ProtectionLevel { get; private set; }
+
+        /// <summary>
+        ///     Friendly name to describe the key with in the dialog box that appears when the key is accessed,
+        ///     null for default name
+        /// </summary>
+        public string FriendlyName { get; private set; }
+
+        /// <summary>
+        ///     Description text displayed in the dialog box when the key is accessed, null for the default text
+        /// </summary>
+        public string Description { get; private set; }
+
+        /// <summary>
+        ///     Description of how the key will be used
+        /// </summary>
+        public string UseContext { get; private set; }
+
+        /// <summary>
+        ///     Title of the dialog box displayed when a newly created key is finalized, null for the default title
```

Comment:
Here’s the MSDN doc for the native CNG version of this struct member. Not sure it’s an improvement. The docs on the BCL crypto apis in general assume a knowledge of the underlying Windows crypto apis.


A pointer to a null-terminated Unicode string that contains the text that will be used in the title of the strong key dialog box when the key is completed. If this member is NULL, a default creation title will be used in the strong key dialog box. This member is only used on key finalization.


From: Jeremy Barton [mailto:notifications@github.com]
Sent: Tuesday, August 4, 2015 4:02 PM
To: dotnet/corefx <corefx@noreply.github.com>
Cc: Atsushi Kanamori <atsushik@microsoft.com>
Subject: Re: [corefx] Implement System.Security.Cryptography.CngKey (#2615)


In src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngUIPolicy.cs<https://github.com/dotnet/corefx/pull/2615#discussion_r36252917>:

> +        ///     null for default name

> +        /// </summary>

> +        public string FriendlyName { get; private set; }

> +

> +        /// <summary>

> +        ///     Description text displayed in the dialog box when the key is accessed, null for the default text

> +        /// </summary>

> +        public string Description { get; private set; }

> +

> +        /// <summary>

> +        ///     Description of how the key will be used

> +        /// </summary>

> +        public string UseContext { get; private set; }

> +

> +        /// <summary>

> +        ///     Title of the dialog box displayed when a newly created key is finalized, null for the default title

I'm not sure what "finalized" means in this context...

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corefx/pull/2615/files#r36252917>.